bigtable/client/row_reader_test.cc

```diff
@@ -634,6 +698,66 @@ TEST_F(RowReaderTest, BeginThrowsAfterImmediateCancel) {
 }
 #endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
 
+TEST_F(RowReaderTest, BeginThrowsAfterCancelClosesStreamNoExcept) {
+  auto parser = bigtable::internal::make_unique<ReadRowsParserMock>();
+  parser->SetRows({"r1"});
+  auto* stream = new MockResponseStream();
+  {
+    testing::InSequence s;
+    EXPECT_CALL(*bigtable_stub_, ReadRowsRaw(_, _)).WillOnce(Return(stream));
+    EXPECT_CALL(*stream, Read(_)).WillOnce(Return(true));
+    EXPECT_CALL(*stream, Read(_)).WillOnce(Return(true));
+    EXPECT_CALL(*stream, Read(_)).WillOnce(Return(true));
+    EXPECT_CALL(*stream, Read(_)).WillOnce(Return(false));
+    EXPECT_CALL(*stream, Finish()).WillOnce(Return(grpc::Status::OK));
+  }
+
+  parser_factory_->AddParser(std::move(parser));
+  bigtable::RowReader reader(
+      client_, "", bigtable::RowSet(), bigtable::RowReader::NO_ROWS_LIMIT,
+      bigtable::Filter::PassAllFilter(), std::move(retry_policy_),
+      std::move(backoff_policy_), metadata_update_policy_,
+      std::move(parser_factory_), false);
+
+  auto it = reader.begin();
+  EXPECT_NE(it, reader.end());
+  EXPECT_EQ(it->row_key(), "r1");
+  EXPECT_NE(it, reader.end());
+  // Manually cancel the call.
+  reader.Cancel();
+  reader.begin();
+  grpc::Status status = reader.Finish();
+  EXPECT_FALSE(status.ok());
+}
+
+TEST_F(RowReaderTest, BeginThrowsAfterImmediateCancelNoExcept) {
+  bigtable::RowReader reader(
+      client_, "", bigtable::RowSet(), bigtable::RowReader::NO_ROWS_LIMIT,
+      bigtable::Filter::PassAllFilter(), std::move(retry_policy_),
+      std::move(backoff_policy_), metadata_update_policy_,
+      std::move(parser_factory_), false);
+  // Manually cancel the call before a stream was created.
+  reader.Cancel();
+
+  reader.begin();
+  grpc::Status status = reader.Finish();
+  EXPECT_FALSE(status.ok());
+}
+
+TEST_F(RowReaderTest, NoExceptionShouldExpectDeathIfErrorIsNotRetrieved) {
+  EXPECT_DEATH_IF_SUPPORTED(
```

Comment:
@coryan added this test case if error is not retrieved

Reply:
Thanks, that works.  Consider:

```C++
std::unique_ptr<RowReader> reader(...);
reader->Cancel();
reader->begin();
EXPECT_DEATH_IF_SUPPORTED(reader.reset(), "");
reader->Finish();
```

Which more narrowly tests for "the destructor will raise".